lib/method_source/code_helpers.rb

```diff
@@ -71,8 +71,8 @@ def complete_expression?(str)
         eval("BEGIN{throw :valid}\n#{str}")
       end
 
-      # Assert that a line which ends with a , or \ is incomplete.
-      str !~ /[,\\]\s*\z/
+      # Assert that a line which ends with a , or \ but not with $\ is incomplete.
+      str !~ /(,|[^\$]\\)\s*\z/
```

Comment:
`MethodSource.complete_expression?('$$')` was true before this patch and is also true after this patch.